htscodecs/utils.h

```diff
@@ -38,6 +38,21 @@
 #include <stdlib.h>
 #include <math.h>
 
+#if defined(__GNUC__) || defined(__clang__)
+#  if !defined(__clang__) && __GNUC__ >= 100
+     // better still on gcc10 for O1 decode of old rans 4x8
+     // gcc 10=246/205 11=243/205 12=230/197
+#    define likely(x)      __builtin_expect_with_probability((x), 1, 0.99)
```

Comment:
clang also has [`__builtin_expect_with_probability`](https://clang.llvm.org/docs/LanguageExtensions.html#builtin-expect-with-probability).  According to the manual, you should be able to query it with something like:
```
#if defined(__has_builtin)
#  if __has_builtin(__builtin_expect_with_probability)
#     define likely(x) __builtin_expect_with_probability((x), 1, 0.99)
#  endif
#endif
```

Reply:
Bah, results appear just wildly different today to the last time I benchmarked.  I hate trying to benchmark things on modern systems!  Just so vague...  Will do a complete battery of tests again on a modern system.  Tedious, but automated at least.